hashintel/hash
# Fix broken Table entity type page

## 🌟 What is the purpose of this PR?



This fixes a bug with the Table entity type page ~and improves block insertion UX by replacing the current block if slash menu is triggered in an empty block.~

## 🔗 Related links





- [Asana task](link) _(internal)_

## 🔍 What does this change?




- SchemaPropertyRow didn't previously account for constraints with values that aren't strings (e.g objects, arrays). This fixes that by converting the values to strings and rendering in a `<pre>` tag for readability.
- ~`onBlockSuggesterChange` logic is updated  to replace the current block if slash menu was triggered in an empty block.~

## 📜 Does this require a change to the docs?



- No

## ❓ How to test this?



1.  Checkout the branch 
1.  Go to Table entity type page and confirm it doesn't crash
1.  Trigger slash (/) menu in an empty block and select a block
1.  Confirm the current block gets replaced instead of a new block being inserted

## 📹 Demo



Before

https://user-images.githubusercontent.com/16928688/178615377-69313a65-14b1-4a7a-b36a-ae89623f313b.mov


After

https://user-images.githubusercontent.com/16928688/178615430-08b41bbf-5662-4547-bbc3-d5c4c40ec25a.mov




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202567514360917
  - https://app.asana.com/0/0/1202500568584729
